bosh/ruby-app-release/unit-tests/spec/helpers.rb

```diff
@@ -0,0 +1,52 @@
+OPEN_TOKEN = /.+\)/
+CLOSE_TOKEN = /^\s*\;\;\s*$/
+CASE_TOKEN = /\s*case.*/
+
+def check_block_formations(rendered_template)
+
+  line_index = 0
+  tokens = Array.new #acts like a stack
+  
+  #we only keep three types of tokens in the stack
+  rendered_template.each_line do |line|
+    case
+    when line.match?(CASE_TOKEN)
+      tokens.push(CASE_TOKEN)
+    when line.match?(OPEN_TOKEN)
+      tokens.push(OPEN_TOKEN)
+    when line.match?(CLOSE_TOKEN)
+      tokens.push(CLOSE_TOKEN)
+    end
+  end
+
+  # remove header token and check if it starts a CASE..ESAC block
+  # if not, then the rest of the tokens are not within a CASE..ESAC block, which is not allowed
+  case_token = tokens.shift
+  return false unless case_token == CASE_TOKEN
+  
+  remaining_tokens_count = tokens.length
+  next_token = CLOSE_TOKEN
+  # pop in LIFO order and see whether tokens are matched
+  # i.e if the current token is OPEN, the next one must be CLOSE. 
+  for i in 1..remaining_tokens_count do
```

Comment:
Using your idea of `.times` would not work here @CedrikHeusser. The reason is that I am modifying the array during the process; I remove elements from it one by one until it is totally consumed. 

I could have made this more performant but it would make it less easy to read.